### PR TITLE
rgw/examples: adding examples for boto3 extensions to AWS S3

### PR DESCRIPTION
**Pull request by yuvalif**
Signed-off-by: Yuval Lifshitz <yuvalif@yahoo.com>

RGW extends the standard (AWS) S3 implementation for some of the APIs.
To make it easier to tests and integrate sample python code was added, together with a json file that allows extending boto3 without any modification to the boto3 source code.

Other existing extension, as well as future extensions, should be added to the json file together with a matching python code.

Documentation added under: https://docs.ceph.com/docs/master/radosgw/s3/python/

### CODE REVIEW

**Comment by yuvalif** [comment]
jenkins test make check

**Comment by theanalyst** [comment]
In addition to boto3, these seamlessly work with aws-cli as well (which uses botocore) so this can be additionally mentioned in readme. Later down the road we can consider packaging this as a pip package or so, so that we can install this as an extension and be used as a simple install for people wanting to use python/aws cli against RGW

File: examples/boto3/README.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+For the standard boto3 client to support these extensions, the: ``service-2.sdk-extras.json`` file should be placed under: ``~/.aws/models/s3/2006-03-01/`` directory.
+For more information see [here](https://github.com/boto/botocore/blob/develop/botocore/loaders.py#L33).
+
+# Developers
+Anyone developing an extension to the S3 API supported by AWS, please modify ``service-2.sdk-extras.json`` (all extensions should go into the same file), so that boto3 could be used to test the new API. 
+In addition, python files with code samples should be added to this directory demonstrating use of the new API.
+When testing you changes please:
+- make sure that the modified file is in the boto3 path as explained above
+- make sure that the standard S3 tests suit is not broken, even with the extensions files in the path
+
```
Comment (theanalyst) [inline examples/boto3/README.md]: In addition to boto, this allows aws cli to use these commands as well. Please make sure to add a useful help text while adding the json fields as these will be directly used by the cli.

Comment (theanalyst) [reply examples/boto3/README.md]: In examples a simple example like 

`aws s3api list-objects --bucket=bucket1 --allow-unordered`

Comment (yuvalif) [reply examples/boto3/README.md]: will test that for the different extensions and update

Comment (yuvalif) [reply examples/boto3/README.md]: @theanalyst do you think we should rename the directory, or would "boto3" make sense in both the cases?

Comment (theanalyst) [reply examples/boto3/README.md]: I think it should be fine

Comment (yuvalif) [reply examples/boto3/README.md]: added note and 3 examples regarding "aws cli" in the readme

File: examples/boto3/README.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
+# Introduction
+This directory contains examples on how to use AWS CLI/boto3 to execrsize thew RadosGW extensions to the S3 API.
+
+# Users
+For the standard client to support these extensions, the: ``service-2.sdk-extras.json`` file should be placed under: ``~/.aws/models/s3/2006-03-01/`` directory.
+For more information see [here](https://github.com/boto/botocore/blob/develop/botocore/loaders.py#L33).
+## Python
+The [boto3 client](https://boto3.amazonaws.com/v1/documentation/api/latest/index.html) could be used with the extensions, code sampels exists in this directory.
```
Comment (cbodley) [inline examples/boto3/README.md]: sampels->samples

File: examples/boto3/README.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
+# Introduction
+This directory contains examples on how to use AWS CLI/boto3 to execrsize thew RadosGW extensions to the S3 API.
```
Comment (cbodley) [inline examples/boto3/README.md]: > execrsize thew

exercise the

File: examples/boto3/README.md (hunk 2 above)
Comment (yuvalif) [reply examples/boto3/README.md]: fixed

File: examples/boto3/README.md (hunk 3 above)
Comment (yuvalif) [reply examples/boto3/README.md]: fixed

**Comment by cbodley** [comment]
good to merge once make check checks!

**Pull request merged**